OpInfos for new_blah functions and some _like functions

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67357 OpInfos for new_blah functions and some _like functions**

This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

Differential Revision: [D31969533](https://our.internmc.facebook.com/intern/diff/D31969533)

zou3519 committed:
OpInfos for new_blah functions and some _like functions

This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

[ghstack-poisoned]

zou3519 committed:
Update on "OpInfos for new_blah functions and some _like functions"


This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

Differential Revision: [D31969533](https://our.internmc.facebook.com/intern/diff/D31969533)

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -2092,6 +2092,55 @@ def sample_inputs_like_fns(self, device, dtype, requires_grad, **kwargs):
+
+    for sample in sample_inputs_like_fns(self, device, dtype, requires_grad, **kwargs):
+        # With high
+        samples.append(SampleInput(sample.input, args=(high,) + sample.args, kwargs=sample.kwargs))
```
Reusing the same tensor across SampleInputs is going to be a problem. These probably need something like `.detach().clone().requires_grad_(requires_grad)` (which can be put in a helper because it's very verbose). Also I suggest rebasing this to pick up the latest OpInfo tests which are more likely to fail if SampleInputs share tensors.

mruberry reviewed (approved):
Super cool, @zou3519.

I made one inline comment that does need to be addressed, however. Basically that SampleInputs need to be completely independent of each other. I also suggest rebasing this because test_noncontiguous_samples will do some independence checking of SampleInputs.

zou3519 committed:
Update on "OpInfos for new_blah functions and some _like functions"


This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

Differential Revision: [D31969533](https://our.internmc.facebook.com/intern/diff/D31969533)

[ghstack-poisoned]

zou3519 committed:
Update on "OpInfos for new_blah functions and some _like functions"


This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

Differential Revision: [D31969533](https://our.internmc.facebook.com/intern/diff/D31969533)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31969533).

mruberry commented:
Ping @zou3519 -- I think this just needs a small tweak for the tests and it's good to go. We should get it in sooner rather than later so it doesn't need to be rebased again and again.

zou3519 commented:
> Ping @zou3519 -- I think this just needs a small tweak for the tests and it's good to go. We should get it in sooner rather than later so it doesn't need to be rebased again and again.

Yup, thanks for the ping

zou3519 commented:
@mruberry one complaint I have about OpInfos is actually the frequent rebases. Someone might put up an OpInfo PR, but sometimes you didn't know there was a test that uses OpInfos that you had to skip. But you only really get this signal 5+ hours later, so on the next day you fix that test and then do a rebase. The "fixing the test and doing a rebase" takes ~20 minutes for me due to needing to resolve merge conflicts, and then maybe I forgot to fix some test and this process repeats for a couple of days.

The root cause seem to be not understanding what tests use OpInfo that one should test locally. I wish there were some sort of a "python run_all_related_tests_for_opinfo.py new_full new_empty ..." script that let me run all the OpInfo tests for a given OpInfo.

zou3519 committed:
Update on "OpInfos for new_blah functions and some _like functions"


This PR adds OpInfos for:
- new_ones, new_zeros, new_full, new_empty
- rand_like, randint_like

I forgot to add the _like functions in a previous PR, so here they are.

Test Plan:
- wait for tests

Differential Revision: [D31969533](https://our.internmc.facebook.com/intern/diff/D31969533)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31969533).
